pkg/kubelet/cm/pod_container_manager_linux.go

```diff
@@ -297,6 +297,31 @@ func (m *podContainerManagerImpl) GetAllPodsFromCgroups() (map[types.UID]CgroupN
 	return foundPods, nil
 }
 
+func (m *podContainerManagerImpl) DoNotEnforceCPULimits() {
+	m.enforceCPULimits = false
```

Comment:
I'm having trouble figuring out why we need this function. can you give an explanation?

Reply:
I needed a way to disable the cpu limits from Kubelet.SyncPod (see below). The PodContainerManager already has the implementation for it (enforceCPULimits), but it was not accessible outside of the module.

So there might be another way to do this, but this was the cleanest and least invasive way I saw.

Reply:
One of the flows (first regular container and then a guaranteed one) is ok with removing the quota when the second container starts. But when the first container is guaranteed I wanted to avoid the two step configuration and then removal of the quota too.

Reply:
recent PR from @haircommander changes a related code path and I think we can evaluate an alternative approach changing (mostly?) `getCgroupConfig` https://github.com/kubernetes/kubernetes/pull/125923

The only possible concern is proper integration with pod workers, will need another pass